testsuite/MDAnalysisTests/topology/test_tprparser.py

```diff
@@ -162,11 +88,9 @@ class TPRDouble(TPRAttrs):
                             + ['SOL'] * 4284,
                             dtype=object)
 
-class TestTPR455Double(TPRDouble):
-
-    __test__ = True
-
-    filename = TPR455Double
+    @pytest.fixture()
+    def filename(self):
+        return TPR455Double
 
 
 class TPR46xBase(TPRAttrs):
```

Comment:
Could we instead `parametrize` the `ref_filename` inside the class?

```python
@pytest.mark.parametrize('ref_filename', [TPR460, TPR461])
class TPR46xBase()
```

Reply:
No. Pytest then expects every method in the class to request a parameter called `ref_filename`.

Reply:
```python
import pytest

@pytest.mark.parametrize('a', range(10), indirect=True)
class Test_A(object):
    def test_foo(self, a):
        assert True

    def test_bar(self):
        assert True
```

This fill fail collection since `test_bar` doesn't require a fixture called `a`